packages/block-editor/src/components/block-list/block.js

```diff
@@ -234,10 +233,11 @@ function BlockListBlock( {
 		clientId,
 		className: classnames(
 			{
-				'is-content-locked': isContentLocking,
+				'is-editing-disabled':
+					blockEditingMode === 'disabled' ||
+					( hasContentLockedParent && ! isContentBlock ),
```

Comment:
Something I'm not sure about is whether or not to have `getEditingMode` contain the template lock logic as well so that components only need to call `getBlockEditingMode` instead of both ``getBlockEditingMode` and `getTemplateLock`.

Reply:
> It makes the selector logic more complicated, but means that blocks/consumers only have to worry about calling one method and adjusting the UI accordingly.

If the API goes public my vote is for this. Maybe it'll be make it easier to maintain too if the conditions are all in one spot?

Reply:
I've put the `contentOnly` template lock logic into `getBlockEditingMode` in one of my more recent commits. I think it's definitely nice having only one selector to call, and hopefully the JSDoc comments I help demystify it. What do you think?

Respecting `canEditBlock` is an interesting idea. I'll investigate. From memory `canEditBlock` is a bit weird and we only use it in the navigation block.

Reply:
OK so you can test what `canEditBlock` does yourself by selecting a Navigation block, open the settings menu, selecting _Lock_, then enabling _Lock editing_.

When you do this you'll see that you can select the Navigation block but that there's an overlay meaning you can't select its children or edit anything. This is different from either `'disabled'` which prevents section altogether or `'contentOnly'` which allows editing but hides toolbars, etc.

I suppose we could add a fourth mode called e.g. `'overlay'` to `getBlockEditingMode` which encapsulates this behaviour. I'm not really sure there's any benefit to doing this though. `getBlockEditingMode` is a private API so it's something we can change easily in the future if there's a need.